clients/drcachesim/common/options.cpp

```diff
@@ -851,11 +851,18 @@ droption_t<uint64_t> op_sched_blocking_switch_us(
     "maybe-blocking to incur a context switch. Applies to -core_sharded and "
     "-core_serial. ");
 
-droption_t<double>
-    op_sched_block_scale(DROPTION_SCOPE_ALL, "sched_block_scale", 1.,
-                         "Input block time scale factor",
-                         "A higher value here results in blocking syscalls "
-                         "keeping inputs unscheduled for longer.");
+droption_t<double> op_sched_block_scale(
+    DROPTION_SCOPE_ALL, "sched_block_scale", 1000., "Input block time scale factor",
+    "The scale applied to the microsecond latency of blocking system calls.  A higher "
+    "value here results in blocking syscalls keeping inputs unscheduled for longer.  "
+    "This should roughly equal the slowdown of instruction record processing versus the "
+    "original (untraced) application execution.");
+
+droption_t<uint64_t> op_sched_block_max_us(DROPTION_SCOPE_ALL, "sched_block_max_us",
+                                           25000000,
```

Comment:
How was the default decided?

Reply:
I see there's a comment at clients/drcachesim/scheduler/scheduler.cpp:1681. Maybe repeat some of it here?

Reply:
Do we need this to be a flag? Or can it be a constant.

Reply:
Added the comment you pointed at, plus added a statement about needing to tweak this for different workloads and simulator speeds, and a statement about the default being picked while tuning a 1-minute-long schedule_stats run on a 112-core 500-thread large application to produce good cpu usage without unduly increasing tool runtime.